File: app/local_plugins/agent_sts_chat/android/src/main/kotlin/com/aiagent/agent_sts_chat/StsChatAgentFlutterPlugin.kt
package com.aiagent.agent_sts_chat

import android.util.Log
import com.aiagent.plugin_interface.NativeAgentRegistry
import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * StsChatAgentFlutterPlugin — STS Chat Agent Flutter 插件入口
 *
 * 在 onAttachedToEngine 时注册 "sts" Agent 类型到 NativeAgentRegistry。
 * agents_server 通过 NativeAgentRegistry.create("sts") 创建 StsChatAgentSession 实例。
 */
class StsChatAgentFlutterPlugin : FlutterPlugin {

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        NativeAgentRegistry.register("sts") { StsChatAgentSession() }
        Log.d("StsChatAgentPlugin", "Registered NativeAgent type=sts")
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: apps/client/local_plugins/agents/agent_sts_chat/ios/Classes/StsChatAgentFlutterPlugin.swift
import Flutter
import ai_plugin_interface
import os.log

/// STS chat agent Flutter plugin entry.
///
/// Registers a factory for the `"sts-chat"` agent type with the global
/// `NativeAgentRegistry`. `agents_server` instantiates a session via
/// `NativeAgentRegistry.shared.create("sts-chat")` when the Dart side calls
/// `createAgent(agentType:"sts-chat", …)`.
public class StsChatAgentFlutterPlugin: NSObject, FlutterPlugin {
    private static let log = OSLog(subsystem: "com.aiagent.agent_sts_chat", category: "Plugin")

    public static func register(with registrar: FlutterPluginRegistrar) {
        NativeAgentRegistry.shared.register("sts-chat") { StsChatAgentSession() }
        os_log("Registered NativeAgent type=sts-chat", log: log, type: .debug)
    }
}
